### DISCUSSION TITLE
Executable size

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#1618] [Q&A] [answered]
I'm new to Contiki. When compiling `hello-world` for example  for tmote sky `make TARGET=sky PORT=/dev/ttyUSB0 hello-world.upload` the executable size reported by Linux is 386,788 Bytes. Is that reasonable?

I thought that Contiki executables are in the order of thousands of bytes and expected the simple `hello-world` to be about 1,000 to 2,000 Bytes.

**Comment by joakimeriksson** [comment, marked as answer]
This includes lots of debug info and other things - try running "size" to see the actual size:

```
 arm-none-eabi-size hello-world.sky 
   text	   data	    bss	    dec	    hex	filename
  42389	    314	   6766	  49469	   c13d	hello-world.sky

```

or

```
arm-none-eabi-size -A hello-world.sky
hello-world.sky  :
section            size    addr
.text             41132   16384
.rodata            1225   57516
.data               314    4352
.bss               6764    4666
.noinit               2   11430
.vectors             32   65504
.stab            169560       0
.stabstr         117321       0
.comment             48       0
.debug_aranges      476       0
.debug_info        6103       0
.debug_abbrev      2491       0
.debug_line        3864       0
.debug_frame        592       0
.debug_str         1098       0
.debug_loc         3838       0
.debug_ranges       132       0
Total            354992

```

**Comment by atiselsts** [comment]
To add to Joakim's answer: the hello world application might not look like doing much, but it actually includes the complete IPv6/6LoWPAN network stack with a MAC protocol and a routing protocol. Running the hello-world app on a node makes the node a fully functional 6LoWPAN network device.
